src/ixdat/techniques/ms.py

```diff
@@ -335,15 +335,97 @@ def __init__(
         """
         self.verbose = verbose
         self.l_cap = l_cap
+        self.l_cap_eff = {}
         self.w_cap = w_cap
         self.h_cap = h_cap
         self.p = p
         self.T = T
         self.gas = gas  # TODO: Gas mixture class. This must be a pure gas now.
 
+
+    def calc_l_cap_eff(
+        self, gas=None, n_dot=None, w_cap=None, h_cap=None,  T=None, p=None
+        ):
+        """Calculate gas specific effective length of the capillary in [m]
+            and add {gas:value} to l_cap_eff (dict)
+
+        Uses Equation 4.10 of Daniel's Thesis.
+
+        Args:
+            w_cap (float): capillary width [m], defaults to self.w_cap
+            h_cap (float): capillary height [m], defaults to self.h_cap
+            n_dot (float): direct measure of flux through capillary with gas, defaults to calculated flux 
+            gas (dict or str): the gas in the chip, defaults to self.gas
+            T (float): Temperature [K], if to be updated
+            p (float): pressure [Pa], if to be updated
+        Returns:
+            float: gas specific effective length in [m]
+        """
+
+        if w_cap is None:
+            w_cap = self.w_cap  # capillary width in [m]
+        if h_cap is None:
+            h_cap = self.h_cap  # capillary height in [m]
+        if T is None:
+            T = self.T
+        if p is None:
+            p = self.p
+        if n_dot is None:
+            n_dot = calc_n_dot_0(gas=gas)
```

Comment:
What's the point of this? If n_dot is not given, then it looks like `l_cap` necessarily becomes just `self.l_cap`.
Maybe better to make `n_dot` a required argument rather than a kwarg. It doesn't make sense to call this function without it, right?

Reply:
(Note that in my own code, I way overdo flexibility of function inputs - I forget to remember from the Zen that "there should only be one right way to do things".)